plugins/woocommerce-blocks/docs/third-party-developers/extensibility/register-product-collection.md

```diff
@@ -0,0 +1,235 @@
+          {
+            textAlign: "center",
+            fontSize: "small",
+          },
+        ],
+      ],
+    ],
+  ],
+});
+```
+
+This will create a collection with a heading, product image, and product price. Here is how it will look like:
+
+![image](https://github.com/woocommerce/woocommerce/assets/16707866/3d92c084-91e9-4872-a898-080b4b93afca)
+
+> ![TIP]
+> You can learn more about inner blocks template in the [Inner Blocks](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/nested-blocks-inner-blocks/#template) documentation.
+
+> ![TIP]
+> You can also take a look at how we are defining our core collections at `plugins/woocommerce-blocks/assets/js/blocks/product-collection/collections` directory. Our core collections will also evolve over time.
+
+## Defining a Collection
+
+We will explain important arguments that can be passed to `__experimentalRegisterProductCollection`. For other arguments, you can refer to the [Block Variation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/#defining-a-block-variation) documentation.
+
+A Collection is defined by an object that can contain the following fields:
+
+- `name` (type `string`): A unique and machine-readable collection name. We recommend using the format `<plugin-name>/product-collection/<collection-name>`. Both `<plugin-name>` and `<collection-name>` should consist only of alphanumeric characters and hyphens (e.g., `my-plugin/product-collection/my-collection`).
+- `title` (type `string`): The title of the collection, which will be displayed in various places including the block inserter and collection chooser.
+- `description` (optional, type `string`): A human-readable description of the collection.
+- `innerBlocks` (optional, type `Array[]`): An array of inner blocks that will be added to the collection. If not provided, the default inner blocks will be used.
+- `isDefault`: ⚠️ It's set to `false` for all collections. 3PDs doesn't need to pass this argument.
+- `isActive`: ⚠️ It will be managed by us. 3PDs doesn't need to pass this argument.
+
+## Attributes
+
+Attributes are the properties that define the behavior of the collection. All the attributes are *optional*. Here are some of the important attributes that can be passed to `__experimentalRegisterProductCollection`:
+
+- `query` (type `object`): The query object that defines the query for the collection. It can contain the following fields:
+    - `offset` (type `number`): The number of items to offset the query by.
+    - `order` (type `string`): The order of the query. Possible values are `asc` and `desc`.
+    - `orderBy` (type `string`): The field to order the query by.
+    - `pages` (type `number`): The number of pages to query.
+    - `perPage` (type `number`): The number of products per page.
+    - `search` (type `string`): The search term to query by.
+    - `taxQuery` (type `object`): The tax query to filter the query by.
```

Comment:
Would be nice to have an example here as well similar to `woocommerceAttributes`

Reply:
Makes sense, I will add an example for `taxQuery` as well. Thanks for the suggestion 🙏🏻